sklearn/callback/_progressbar.py

```diff
@@ -0,0 +1,257 @@
+            # First time we discover this directory -> store the computation tree
+            # If the computation tree is not readable yet, skip and try again next time
+            computation_tree = load_computation_tree(this_dir)
+            if computation_tree is None:
+                return
+
+            self._computation_trees[this_dir] = (computation_tree, {})
+
+        computation_tree, task_ids = self._computation_trees[this_dir]
+
+        for node in computation_tree.iterate(include_leaves=True):
+            if node.children:
+                # node is not a leaf, create or update its task
+                if node.tree_status_idx not in task_ids:
+                    visible = True
+                    if (
+                        self.max_depth_show is not None
+                        and depth + node.depth > self.max_depth_show
+                    ):
+                        # If this node is deeper than what we want to show, we create
+                        # the task anyway but make it not visible
+                        visible = False
+
+                    task_ids[node.tree_status_idx] = self._progress_ctx.add_task(
+                        self._format_task_description(node, computation_tree, depth),
+                        total=node.max_iter,
+                        visible=visible,
+                    )
+
+                task_id = task_ids[node.tree_status_idx]
+                task = self._progress_ctx.tasks[task_id]
+                self._progress_ctx._ordered_tasks.append(task)
+
+                parent_task = self._get_parent_task(node, computation_tree, task_ids)
+                if parent_task is not None and parent_task.finished:
+                    # If the task of the parent node is finished, make this task
+                    # finished. It can happen if some computations are stopped
+                    # before reaching max_iter.
+                    visible = True
+                    if (
+                        self.max_depth_keep is not None
+                        and depth + node.depth > self.max_depth_keep
+                    ):
+                        # If this node is deeper than what we want to keep in the output
+                        # make it not visible
+                        visible = False
+                    self._progress_ctx.update(
+                        task_id, completed=node.max_iter, visible=visible, refresh=False
+                    )
+                else:
+                    node_progress = computation_tree.get_progress(node)
+                    if node_progress != task.completed:
+                        self._progress_ctx.update(
+                            task_id, completed=node_progress, refresh=False
+                        )
+            else:
+                # node is a leaf, look for tasks of its sub computation tree before
+                # going to the next node
+                child_dir = this_dir / str(node.tree_status_idx)
```

Comment:
I think we should abstract away the filesystem that backs the computation trees because:

1. I do not think we want third party developers writing Callbacks to worry about the filesystem.
2. It will be easier to switch to another inter-process communication method in the future.

Reply:
That's probably better yes. I'll try to come up with a more friendly solution